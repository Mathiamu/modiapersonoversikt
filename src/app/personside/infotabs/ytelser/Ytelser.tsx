import * as React from 'react';

import YtelseListe from './YtelserListe';
import { ScrollBar } from '../utils/InfoTabsScrollBar';
import styled from 'styled-components';
import ValgtYtelse from './ValgtYtelse';

const Styling = styled.section`
    flex-grow: 1; /* IE11 */
    display: flex;
    > *:first-child {
        min-width: 19rem;
        flex-basis: 19rem;
        flex-grow: 0;
        padding-top: 1rem;
    }
    > *:last-child {
        flex-grow: 1;
    }
    align-items: flex-start;
`;

function Ytelser() {
    return (
        <Styling>
            <ScrollBar keepScrollId="ytelser">
                <YtelseListe />
            </ScrollBar>
            <ScrollBar keepScrollId="ytelser">
                <ValgtYtelse />
            </ScrollBar>
        </Styling>
    );
}

export default Ytelser;
