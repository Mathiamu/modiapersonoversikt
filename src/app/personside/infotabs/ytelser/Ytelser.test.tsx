import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import * as React from 'react';
import Ytelser from './Ytelser';

test('Om Ytelser matcher snapshot', () => {
    const resultat = renderer.create(
        <TestProvider>
            <Ytelser />
        </TestProvider>
    );

    expect(resultat.toJSON()).toMatchSnapshot();
});
