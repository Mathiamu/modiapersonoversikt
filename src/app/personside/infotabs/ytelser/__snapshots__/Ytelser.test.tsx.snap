// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om Ytelser matcher snapshot 1`] = `
.c6 {
  position: relative;
  padding: .5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c6:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

.c6:hover > button:last-child {
  padding-left: 0.5rem;
}

.c6 > *:first-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c7 {
  border: none;
  padding: 0;
  height: 2rem;
  width: 2rem;
  border-radius: .35rem;
  cursor: pointer;
  background-color: transparent;
  -webkit-transition: padding-left 0.3s;
  transition: padding-left 0.3s;
}

.c7 .nav-frontend-chevron {
  width: 1.5625rem;
}

.c7:focus {
  outline: none;
  box-shadow: none;
}

.c7:focus::after {
  position: absolute;
  top: 0;
  left: 0;
  content: '';
  box-shadow: inset 0 0 0 0.1875rem #254b6d;
  height: 100%;
  width: 100%;
  pointer-events: none;
}

.c3 {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 0.5rem rgba(0,0,0,0.1),0 0 0 1px rgba(0,0,0,0.1);
}

.c5 > *:not(:first-child) {
  border-top: solid 0.0625rem #78706a;
}

.c4 {
  padding: 0.9375rem;
  border-bottom: solid 0.0625rem #78706a;
}

.c2 {
  overflow: auto;
  max-height: 100%;
  padding: .5rem;
}

.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:45rem) {
  .c1 {
    max-height: 100%;
    max-width: 100%;
  }
}

@media (min-width:45rem) {
  .c1 {
    height: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }

  .c1 > *:first-child {
    min-width: 19rem;
    -webkit-flex-basis: 19rem;
    -ms-flex-preferred-size: 19rem;
    flex-basis: 19rem;
    -webkit-box-flex: 0.5;
    -webkit-flex-grow: 0.5;
    -ms-flex-positive: 0.5;
    flex-grow: 0.5;
  }

  .c1 > *:last-child {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onScroll={[Function]}
    >
      <section
        className="c3"
      >
        <div
          className="c4"
        >
          <h2
            className="typo-undertittel"
          >
            Ytelser
          </h2>
        </div>
        <nav
          aria-label="Velg ytelser"
        >
          <ol
            className="c5"
            role="tablist"
          >
            <li>
              <div
                className="c6"
                onClick={[Function]}
              >
                <div>
                  <h3
                    className="typo-undertittel"
                  >
                    Sykepengerrettighet
                  </h3>
                  <p
                    className="typo-element"
                  >
                    ID-dato
                  </p>
                  <p
                    className="typo-normal"
                  >
                    06.02.2019
                  </p>
                </div>
                <button
                  aria-label="Vis sykepenger"
                  aria-selected={false}
                  className="c7"
                >
                  <i
                    className="nav-frontend-chevron chevronboks chevron--hoyre"
                  />
                </button>
              </div>
            </li>
            <li>
              <div
                className="c6"
                onClick={[Function]}
              >
                <div>
                  <h3
                    className="typo-undertittel"
                  >
                    Pleiepenger sykt barn
                  </h3>
                  <p
                    className="typo-element"
                  >
                    ID-dato
                  </p>
                  <p
                    className="typo-normal"
                  >
                    14.11.2018
                  </p>
                  <p
                    className="typo-element"
                  >
                    Barnets f.nr: 
                  </p>
                  <p
                    className="typo-normal"
                  >
                    08031815240
                  </p>
                </div>
                <button
                  aria-label="Vis pleiepenger"
                  aria-selected={false}
                  className="c7"
                >
                  <i
                    className="nav-frontend-chevron chevronboks chevron--hoyre"
                  />
                </button>
              </div>
            </li>
            <li>
              <div
                className="c6"
                onClick={[Function]}
              >
                <div>
                  <h3
                    className="typo-undertittel"
                  >
                    Foreldrepenger
                  </h3>
                  <p
                    className="typo-element"
                  >
                    ID-dato
                  </p>
                  <p
                    className="typo-normal"
                  >
                    21.11.2017
                  </p>
                </div>
                <button
                  aria-label="Vis foreldrepenger"
                  aria-selected={false}
                  className="c7"
                >
                  <i
                    className="nav-frontend-chevron chevronboks chevron--hoyre"
                  />
                </button>
              </div>
            </li>
          </ol>
        </nav>
      </section>
    </div>
    <div
      className="c2"
      onScroll={[Function]}
    >
      <div
        className="alertstripe alertstripe--info"
      >
        <span
          aria-label="info"
          className="alertstripe__ikon"
        >
          <svg
            focusable="false"
            height="1.5em"
            kind="info-sirkel-fyll"
            viewBox="0 0 24 24"
            width="1.5em"
          >
            <g
              fill="none"
            >
              <path
                d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
                fill="#669DB4"
              />
              <path
                d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
                fill="#FFF"
              />
            </g>
          </svg>
        </span>
        <div
          className="typo-normal alertstripe__tekst"
        >
          Fant ingen ytelser for bruker
        </div>
      </div>
    </div>
  </div>
</section>
`;
